### PR TITLE
fix: restore extra-cursor selection highlights on the cursor line (#114)

### LINKED ISSUE #114: fix: extra-cursor selection highlights invisible on the cursor line
## Summary

When text like `hello hello hello` appears on a single line and the user selects all occurrences of `hello` using multi-cursor selection (e.g., Ctrl+D / "Select All Occurrences"), the **selection highlight for extra cursors is not visible on the cursor line**. The primary cursor's selection shows correctly, but extra-cursor selections on the same line appear unhighlighted.

## Steps to Reproduce

1. Open a file containing a line like `hello hello hello`
2. Select the first `hello`
3. Use Ctrl+D (or "Add Selection to Next Find Match") to select the next `hello` occurrences
4. Observe: only the primary cursor's selection is highlighted; extra-cursor selections on the same line have no visible background

## Expected Behavior

All selection regions (primary and extra cursors) should be visually highlighted with the selection background color, matching VS Code behavior.

## Root Cause

The rendering order in TextArea's `_render_line_impl()` causes this:

1. **`get_line()`** is called — our override in `MultiCursorTextArea.get_line()` applies `selection_style` to extra-cursor selection ranges via `line.stylize()`
2. **`cursor_line_style`** is applied to the **entire line** — this overwrites the extra-cursor selection backgrounds set in step 1
3. **Primary selection** is applied — this happens **after** `cursor_line_style`, so it survives

The `_inject_extra_cursors()` pipeline stage (added in #105) only restores **cursor positions** (single cells), not the full **selection ranges**. Extra-cursor selections applied in `get_line()` are lost when `cursor_line_style` paints over the entire line.

### Relevant code

- `MultiCursorTextArea.get_line()` — applies extra-cursor selections (line ~1260)
- `MultiCursorTextArea._inject_extra_cursors()` — restores only cursor cells, not selections (line ~606)
- TextArea `_render_line_impl()` — applies `cursor_line_style` after `get_line()`, then primary selection after that

## Proposed Fix

Add a `_inject_extra_selections()` stage to the rendering pipeline (similar to `_inject_extra_cursors()`) that re-applies `selection_style` at extra-cursor selection ranges on the cursor line, after `cursor_line_style` has been applied.

Alternatively, modify `_inject_extra_cursors()` to handle both cursor positions and selection ranges.

## Related Issues

- #105 (CLOSED) — fixed extra **cursor** visibility on the cursor line (same root cause, different symptom)
- #106 (CLOSED) — fixed indentation guides / whitespace invisible on cursor line
- #112 (CLOSED) — refactored overlay colors to theme-aware CSS component classes

All three addressed different aspects of `cursor_line_style` overwriting styles set in `get_line()`. This issue covers the remaining case: extra-cursor **selection ranges**.

### PR DESCRIPTION
**Pull request by rishubil**
> [!NOTE]
> This PR was auto-generated by a [Claude Code](https://claude.com/claude-code) agent.
> All code changes, tests, and this PR description were produced by the agent.

## Summary
- Extra-cursor selection highlights on the cursor line are now visible after `cursor_line_style` overwrite
- `_inject_extra_cursors()` extended to re-apply `selection_style` at extra-cursor selection ranges
- Cursor cells take priority over selection cells when they overlap

## Investigation
The rendering pipeline in `MultiCursorTextArea` applies styles in this order:
1. `get_line()` applies `selection_style` to extra-cursor selection ranges in the `Text` object
2. `TextArea._render_line()` applies `cursor_line_style` (bgcolor) to the entire cursor line, **overwriting** the selection highlights from step 1
3. `_inject_extra_cursors()` restored only `cursor_style` at cursor positions but **not** `selection_style` at selection ranges

This is the same class of bug as #105 (cursor visibility) but for selection ranges instead of cursor cells.

## Plan
Extend `_inject_extra_cursors()` to also handle selection range re-injection:
- Fetch `selection_style` from the theme alongside `cursor_style`
- Look up `_cached_selection_ranges` for the current line and resolve `None` end values
- Apply selection_style at selection range columns in the per-character loop, with cursor_style taking priority

## Changes
- **`multi_cursor_text_area.py`**: Extended `_inject_extra_cursors()` to re-apply `selection_style` at extra-cursor selection ranges. Added early return when both styles are None. Conditional cursor_cols/sel_ranges construction for efficiency.
- **`test_multi_cursor.py`**: Added 3 new tests to `TestExtraCursorVisibility`:
  - `test_extra_cursor_selection_visible_on_cursor_line` — main bug verification
  - `test_extra_cursor_inside_selection_has_cursor_style` — cursor priority edge case
  - `test_extra_cursor_selection_visible_on_non_cursor_line` — regression guard
- **`docs/features/internals.md`**: Updated "Visual rendering" section to document selection range re-injection
- **`CHANGELOG.md`**: Added entry for the bug fix

## Test Plan
- [x] `test_extra_cursor_selection_visible_on_cursor_line` — verifies selection bg differs from plain cursor-line bg
- [x] `test_extra_cursor_inside_selection_has_cursor_style` — verifies cursor_style priority over selection_style
- [x] `test_extra_cursor_selection_visible_on_non_cursor_line` — regression guard for non-cursor lines
- [x] `test_extra_cursor_has_distinct_bg_on_cursor_line` — existing test still passes
- [x] Full test suite: 2189 parallel + 61 snapshot tests all pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/textual_code/widgets/multi_cursor_text_area.py [hunk 1]
```diff
@@ -626,10 +630,31 @@ def _inject_extra_cursors(self, strip: Strip, y: int) -> Strip:
...
+            sel_ranges = []
+            sel_starts = []
+
+        if not cursor_cols and not sel_ranges:
             return strip
 
+        def _in_selection(col: int) -> bool:
+            idx = bisect_right(sel_starts, col) - 1
+            return idx >= 0 and sel_ranges[idx][0] <= col < sel_ranges[idx][1]
+
```
[the remark this answers is not in the document]
Comment (rishubil) [reply src/textual_code/widgets/multi_cursor_text_area.py]: > 🤖 This reply was generated by [Claude Code](https://claude.com/claude-code).

Fixed in a4f89a3. Overlapping ranges are now merged before bisect lookup:

```python
sel_ranges: list[tuple[int, int]] = [merged[0]]
for s, e in merged[1:]:
    prev_s, prev_e = sel_ranges[-1]
    if s <= prev_e:
        sel_ranges[-1] = (prev_s, max(prev_e, e))
    else:
        sel_ranges.append((s, e))
```

This ensures nested ranges like `(2,20)` + `(5,10)` → `(2,20)` are handled correctly by the bisect-based membership check.

File: src/textual_code/widgets/multi_cursor_text_area.py [hunk 2]
```diff
@@ -655,6 +684,8 @@ def _inject_extra_cursors(self, strip: Strip, y: int) -> Strip:
                 content_col = cell_pos - gutter_width + scroll_x
                 if content_col in cursor_cols:
                     new_segments.append(Segment(ch, cursor_style))
+                elif sel_ranges and _in_selection(content_col):
+                    new_segments.append(Segment(ch, selection_style))
                 else:
                     new_segments.append(Segment(ch, seg_style))
                 cell_pos += cell_len(ch)
```
[the remark this answers is not in the document]
Comment (rishubil) [reply src/textual_code/widgets/multi_cursor_text_area.py]: > 🤖 This reply was generated by [Claude Code](https://claude.com/claude-code).

This is a pre-existing issue that also affects the existing `cursor_style` injection (line 685: `if content_col in cursor_cols`), which was introduced in #105. Both cursor and selection comparisons use the same cell-based coordinate system. Fixing this would require a broader refactor of the entire `_inject_extra_cursors` method and is out of scope for this PR.

File: src/textual_code/widgets/multi_cursor_text_area.py [hunk 3]
```diff
@@ -626,10 +630,42 @@ def _inject_extra_cursors(self, strip: Strip, y: int) -> Strip:
...
+            if cursor_style
+            else set()
+        )
+
+        # Only re-inject selection ranges on the cursor line where
+        # cursor_line_style overwrites get_line() styling.  On other lines
+        # get_line()'s Text.stylize() layering preserves syntax highlighting.
+        is_cursor_line = line_index == self.cursor_location[0]
+        raw_ranges = self._cached_selection_ranges.get(line_index) or ()
+        if selection_style and raw_ranges and is_cursor_line:
```
[the remark this answers is not in the document]
Comment (rishubil) [reply src/textual_code/widgets/multi_cursor_text_area.py]: > 🤖 This reply was generated by [Claude Code](https://claude.com/claude-code).

Good catch! Fixed in a9e73bf. Now using `self.highlight_cursor_line and self._has_cursor and line_index == self.cursor_location[0]` to match the exact conditions under which `TextArea._render_line()` applies `cursor_line_style`.
